src/api/internal.rs

```diff
@@ -1288,16 +1288,20 @@ impl<T: Pixel> ContextInner<T> {
       return Err(EncoderStatus::Failure);
     }
 
-    let mut psnr = None;
-    if self.config.show_psnr {
-      if let Some(ref rec) = rec {
-        let original_frame = self.get_frame(input_frameno);
-        psnr = Some(calculate_frame_psnr(&*original_frame, rec, bit_depth));
-      }
+    let mut metrics = QualityMetrics::default();
```

Comment:
This would always compute the metrics. It should not, and could be argued the feature should be removed from the encoder.